packages/app/src/cli/services/function/build.ts

```diff
@@ -137,3 +157,121 @@ export async function runFunctionRunner(fun: ExtensionInstance, options: Functio
     stderr: 'inherit',
   })
 }
+
+export interface JavyBuilder {
+  bundle(
+    fun: ExtensionInstance<FunctionConfigType>,
+    options: JSFunctionBuildOptions,
+  ): Promise<BuildResult<BuildOptions>>
+  compile(fun: ExtensionInstance<FunctionConfigType>, options: JSFunctionBuildOptions): Promise<void>
+}
+
+export const DefaultJavyBuilder: JavyBuilder = {
+  async bundle(fun: ExtensionInstance<FunctionConfigType>, options: JSFunctionBuildOptions) {
+    return bundleExtension(fun, options)
+  },
+
+  async compile(fun: ExtensionInstance<FunctionConfigType>, options: JSFunctionBuildOptions) {
+    return runJavy(fun, options)
+  },
+}
+
+const JAVY_WORLD = 'shopify-function'
+export class ExportJavyBuilder implements JavyBuilder {
+  exports: string[]
+
+  constructor(exports: string[]) {
+    this.exports = exports
+  }
+
+  async bundle(fun: ExtensionInstance<FunctionConfigType>, _options: JSFunctionBuildOptions) {
+    if (!fun.entrySourceFilePath) {
+      throw new Error('Could not find your function entry point. It must be in src/index.js or src/index.ts')
+    }
+
+    const contents = this.entrypointContents
+    outputDebug('Generating dist/function.js using generated module:')
+    outputDebug(contents)
+
+    const esbuildOptions: Parameters<typeof esBuild>[0] = {
+      outfile: joinPath(fun.directory, 'dist/function.js'),
+      stdin: {
+        contents,
+        loader: 'ts',
+        resolveDir: fun.directory,
+      },
+      alias: {
+        'user-function': fun.entrySourceFilePath,
+      },
+      logLevel: 'silent',
```

Comment:
Will setting the log level to silent also silence what's causing the error if there is one or will that continue to be emitted?

Reply:
I don't know the answer but that behaviour isn't changing in this PR, right? 😅

```javascript
function getESBuildOptions(directory: string, entryPoint: string, userFunction: string): Parameters<typeof esBuild>[0] {
  const esbuildOptions: Parameters<typeof esBuild>[0] = {
    outfile: joinPath(directory, 'dist/function.js'),
    entryPoints: [entryPoint],
    alias: {
      'user-function': userFunction,
    },
    logLevel: 'silent',
    bundle: true,
    legalComments: 'none',
    target: 'es2022',
    format: 'esm',
  }
  return esbuildOptions
}
```

Jimmy mentioned maybe refactoring the previous build stuff, will see if I have time 🙈

Reply:
Ah, the diff in its present state made that difficult to see. I'll resolve this.